### PR TITLE
Demonstrating caching of attributes and associations

### LINKED ISSUE #1: None of the serializers invoke caching... and there's no AMS initializer... is this incomplete?

### PR DESCRIPTION
**Pull request by bf4**
It is perhaps surprising that a serializer only cached its attributes,
not its associations? (At least in this case)

Closes #1

### CODE REVIEW

File: app/controllers/api/v1/foos_controller.rb [hunk 1]
```diff
@@ -22,6 +25,37 @@ def foo2
           f.bars << Bar.new
         end
       end
+
+      def before_action
+        case params[:cache]
+        when 'clear'.freeze then ActionController::Base.cache_store.clear
+        when 'on'.freeze    then self.perform_caching = true # NOTE: AMS caching is not changed by this
+        when 'off'.freeze   then self.perform_caching = false
```
Comment (bf4) [inline app/controllers/api/v1/foos_controller.rb:33]: just for convenience

File: app/serializers/api/v1/bar_serializer.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module Api
   module V1
     class BarSerializer < ActiveModel::Serializer
+      cache
```
Comment (bf4) [inline app/serializers/api/v1/bar_serializer.rb:4]: ❗️ required to cache the bars that foo has

File: config/environments/development.rb [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+  config.eager_load = true
+  config.cache_classes = true
+  config.action_controller.perform_caching = true
+  config.action_controller.cache_store = :dalli_store
+  config.consider_all_requests_local = false
+  # Configure logging
+  ActiveSupport::Cache::Store.logger = config.logger
+  config.log_level = :debug
+  # otherwise deadlocks can occurr
+  config.middleware.delete 'Rack::Lock'
```
Comment (bf4) [inline config/environments/development.rb:12]: Not sure what env you were running in; I assume dev. So this, makes it behave like prod.

File: app/models/bar.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 class Bar < ActiveRecord::Base
   def batz
+    Rails.logger.info "Bar#batz[#{id}]: I'm going to sleep."
     sleep 1
```
Comment (bf4) [inline app/models/bar.rb:4]: 💤

**Comment by metaskills** [comment]
Thanks! So the basic working here was to make sure `action_controller.perform_caching = true` was set in development (forgot about that, too used to Rails v5 dev:cache setup) and then there is the gotcha that associations are not cached. Yes, that does surprise me. Appreciate you making this PR and highlighting that.

**Comment by bf4** [comment]
Sure.. keep me in the loop with what you learn.  The attributes vs associations distinction both makes sense and surprised me... need to think about

**Pull request closed without merging**